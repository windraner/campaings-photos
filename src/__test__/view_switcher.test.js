import React from 'react';
import ViewSwitcher from '../components/main_container/top_bar/view_switcher/view_switcher';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

const mock = {
  itemsView: 'list',
  stateHandler() {},
};

describe('ViewSwitcher', () => {
  it('matches the snapshot', () => {
    const tree = shallow(<ViewSwitcher {...mock} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
  it('matches the snapshot', () => {
    const tree = shallow(<ViewSwitcher {...mock} itemsView={'grid'} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
