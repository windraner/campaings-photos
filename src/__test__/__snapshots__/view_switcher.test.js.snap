// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewSwitcher matches the snapshot 1`] = `
<Fragment>
  <div
    className="campaigns-photos__viewSwitcher-grid d-flex align-items-center justify-content-center ml-3"
    onClick={[Function]}
    style={
      Object {
        "color": "#b7c0cd",
      }
    }
  />
  <div
    className="campaigns-photos__viewSwitcher-list d-flex flex-column justify-content-between"
    onClick={[Function]}
  >
    <span
      className="campaigns-photos__viewSwitcher-line"
      style={
        Object {
          "background": "#f67646",
        }
      }
    />
    <span
      className="campaigns-photos__viewSwitcher-line"
      style={
        Object {
          "background": "#f67646",
        }
      }
    />
    <span
      className="campaigns-photos__viewSwitcher-line"
      style={
        Object {
          "background": "#f67646",
        }
      }
    />
  </div>
</Fragment>
`;

exports[`ViewSwitcher matches the snapshot 2`] = `
<Fragment>
  <div
    className="campaigns-photos__viewSwitcher-grid d-flex align-items-center justify-content-center ml-3"
    onClick={[Function]}
    style={
      Object {
        "color": "#f67646",
      }
    }
  />
  <div
    className="campaigns-photos__viewSwitcher-list d-flex flex-column justify-content-between"
    onClick={[Function]}
  >
    <span
      className="campaigns-photos__viewSwitcher-line"
      style={
        Object {
          "background": "#b7c0cd",
        }
      }
    />
    <span
      className="campaigns-photos__viewSwitcher-line"
      style={
        Object {
          "background": "#b7c0cd",
        }
      }
    />
    <span
      className="campaigns-photos__viewSwitcher-line"
      style={
        Object {
          "background": "#b7c0cd",
        }
      }
    />
  </div>
</Fragment>
`;
